### DISCUSSION TITLE
`VAULT_TOKEN` not read when using vault secretBackend?

### DISCUSSION THREAD

**Discussion by V0lantis** [apache/airflow#37336] [Q&A]
Hello,

I am trying to set up vault `secretBackend` locally. 
From the [documentation](https://airflow.apache.org/docs/apache-airflow-providers-hashicorp/stable/secrets-backends/hashicorp-vault.html#storing-and-retrieving-connections-using-connection-uri-representation:~:text=You%20can%20also%20set%20and%20pass%20values%20to%20Vault%20client%20by%20setting%20environment%20variables.%20All%20the%20environment%20variables%20listed%20at%20https%3A//www.vaultproject.io/docs/commands/%23environment%2Dvariables%20are%20supported.), it seems that Airflow vault provider supports any environment variables which are defined, including `VAULT_TOKEN`.

> You can also set and pass values to Vault client by setting environment variables. All the environment variables listed at https://www.vaultproject.io/docs/commands/#environment-variables are supported.

Here is my configuration :

```cfg
backend = airflow.providers.hashicorp.secrets.vault.VaultBackend

backend_kwargs = {
    "connections_path": "connections",
    "variables_path": "variables",
    "url": "http://127.0.0.1:8200",
    "mount_point": "airflow"
    }
```
I have tried to launch `airflow connections list` of a get, but debugging the code seem to tell me that airflow never tries to read [the `VAULT_TOKEN` env variable.](https://github.com/apache/airflow/blob/main/airflow/providers/hashicorp/_internal_client/vault_client.py#L128-L129)

Any idea what I am doing wrong?

**Comment by V0lantis** [comment]
I opened a [PR](https://github.com/apache/airflow/pull/37337). Feel free to discuss my proposition. I guess, if this feature wasn't incorporated before, there might be a reason that I couldn't see now. Happy to exchange upon it.

**Comment by andreyolv** [comment]
Add fields "auth_type" and "token" in the backend_kwargs:
Example:
'{"connections_path": "connections", "variables_path": "variables", "mount_point": "airflow", "url": "http://vault.vault:8200", "auth_type": "token", "token": "$VAULT_TOKEN"}'

or if you ar using approle in your vault:
"auth_type": "approle", "role_id":"$VAULT_ROLE_ID", "secret_id":"$VAULT_SECRET_ID"
